### DISCUSSION TITLE
Adapter node breaks my app.

### DISCUSSION THREAD

**Discussion by Germain-L** [sveltejs/kit#9573] [Q&A] [answered]
Evolving this issue : https://github.com/sveltejs/kit/discussions/9562

My app throws errors because of dependencies only when I run it with `node build/index.js` when using [adapter-node](https://kit.svelte.dev/docs/adapter-node). Works fine when using `preview`

I thought it came from mongodb but the error persists when using prisma + postgresql

Here's what happens :

on build :
```
> Using @sveltejs/adapter-node
"mock-aws-s3" is imported by "mock-aws-s3?commonjs-external", but could not be resolved – treating it as an external dependency.
"aws-sdk" is imported by "aws-sdk?commonjs-external", but could not be resolved – treating it as an external dependency.
"nock" is imported by "nock?commonjs-external", but could not be resolved – treating it as an external dependency.
Circular dependency: node_modules/readable-stream/lib/_stream_readable.js -> node_modules/readable-stream/lib/_stream_duplex.js -> node_modules/readable-stream/lib/_stream_readable.js
Circular dependency: node_modules/readable-stream/lib/_stream_duplex.js -> node_modules/readable-stream/lib/_stream_writable.js -> node_modules/readable-stream/lib/_stream_duplex.js
Circular dependency: node_modules/glob/glob.js -> node_modules/glob/sync.js -> node_modules/glob/glob.js
Circular dependency: node_modules/@mapbox/node-pre-gyp/lib/node-pre-gyp.js -> node_modules/@mapbox/node-pre-gyp/lib/pre-binding.js -> node_modules/@mapbox/node-pre-gyp/lib/node-pre-gyp.js
Circular dependency: node_modules/semver/classes/comparator.js -> node_modules/semver/classes/range.js -> node_modules/semver/classes/comparator.js
  ✔ done
✓ built in 3.75s
Done in 4.41s.
```

When ran with `node build/index.js` :

```
Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'mock-aws-s3' imported from /home/gmn/work/test/build/server/chunks/5-4a6e0b61.js
    at new NodeError (node:internal/errors:399:5)
    at packageResolve (node:internal/modules/esm/resolve:889:9)
    at moduleResolve (node:internal/modules/esm/resolve:938:20)
    at defaultResolve (node:internal/modules/esm/resolve:1153:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:838:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:77:40)
    at link (node:internal/modules/esm/module_job:76:36)
```

**Comment by vekunz** [comment]
Can you post your `package.json`, `vite.config.js` and `svelte.config.js?`

**Comment by Germain-L** [comment, marked as answer]
I figured it out, it was the package bcrypt that was causing errors

**Reply by micabal** [reply]
So, what did you do? I have same error. Thanks ;-)

**Reply by SebRM** [reply]
I am also having the same issue and would appreciate a solution if you have one, other than just replacing bcrypt with an alternative.

**Reply by micabal** [reply]
Hi @SebRM, I made it work somehow, but I do not really remember clearly how. Now I have no issues. Try adding the library in your `dependencies` instead of the `devDependencies` in your package.json. Run `npm install` and finally `npm run build`

**Reply by SebRM** [reply]
Alright, thanks. So far I have switched to use [bcryptjs](https://www.npmjs.com/package/bcryptjs) instead which might be slightly slower but works without node-pre-gyp, which seems to be [the issue](https://github.com/mapbox/node-pre-gyp/issues/661).

**Reply by wwmike** [reply]
I have the same. Moving to dependencies causes me another error:
`Error [ERR_MODULE_NOT_FOUND]: Cannot find package '/var/www/html/home/node_modules/bcrypt/' imported from /var/www/html/home/build/server/chunks/3-uNASF2Oy.js`

So you happen to know why it tries to find it in node_modules? I have "type": "module" in the package.json

**Reply by Carlos-err406** [reply]
installing it as a dependency fixed that for me
npm rm bcrypt && npm i bcrypt

**Reply by LitleProgrammer** [reply]
> installing it as a dependency fixed that for me npm rm bcrypt && npm I bcrypt

Can confirm. This fixed the issue for me
